doc: Fix Action documentation (#505)
slsa-framework/slsa-github-generator: .github/actions/README.md

> In particular, *composite actions* need to use inline scripts and cannot invoke script files stored in the builder's repository.

Reviewer: ```suggestion
In particular, *composite actions* need should not invoke script files stored in the git repository. Only inline scripts are permitted under these rules.
```

Reviewer 2: "*composite actions* need should not invoke script" -> "*composite actions* should not invoke script" ?

In general, Actions that need to "checkout" their code should use [Dockerfile](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action) or [nodejs](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action)-type projects "compiled" with [ncc](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github). An example of such an Action
is the `./github/actions/detect-workflow` Action.

There is one exception today: the `./github/actions/generate-builder` Action. It "checkouts" its own code and is allowed to do it
because it does so in a job that never "checkouts" the calling repository. (Note: the code will be migrated to 

> a Dockerfile or nodejs-type projects in the future).

Reviewer 2: ```suggestion
a Dockerfile or nodejs-type project in the future).
```

## Development

To create or update an internal Action, follow the 2 following steps:

1. Create / modify the Action under `./github/actions/<your-action>` and get the changes merged. Let's call the resulting

> commit hash after merge `CH`. (Note: This won't affect builder's behavior since

Reviewer: ```suggestion
commit hash after merge `CH`. (Note: This won't affect any workflow's behavior since
```
